File: Data Structures & Algorithms/binary-search/submission-4.py
class Solution:

    def bs(self, l,r,nums,target) -> int:
        if l > r:
            return -1

        mid = (l + r) // 2

        if nums[mid] == target:
            return mid
        elif nums[mid] > target:
            return self.bs(l, mid - 1, nums, target)
        else:
            return self.bs(mid+1, r, nums, target)

    def search(self, nums: List[int], target: int) -> int:
        return self.bs(0,len(nums) - 1, nums, target)

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-4.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int window = 0; int l = 0; int r = 0; int n = nums.size(); int minsubs = n + 1; 

        // make the first window

        while (r < n) {

            window += nums[r];
            
            // cout << "Window at: " << window << "\t r: " << r << endl;

            // window >= target
            while (window >= target) {
                minsubs = min(minsubs, r-l+1);
                window -= nums[l]; l++;
            }
            r++;
            // cout << "\tWindow at: " << window << "\tl: " << l << "\tr: " << r << endl;
        }

        if (minsubs == n+1) {  return 0;}
        return minsubs;
    }
};